### DISCUSSION TITLE
SimpleExchange dapp Update

### DISCUSSION THREAD

**Discussion by Jorge-Lopes** [Agoric/agoric-sdk#8460] [Q&A]
## Source code (Work in Progress):
https://github.com/alexanderem49/simple-exchange

## Context: 
The purpose of this task is to take advantage of the simpleExchange simplicity to help introduce new developers to the Agoric stack. We are updating and extending the existing component to include the following resources:
- Contracts:
    - upgradable
    - non-upgradable
- Tests
    - unit tests
    - upgrade test
    - integration test
- Deploy on devnet
    - core-eval
    - proposals
    - deployment tests
- UI
- Documentation
    - Component page
    - Tutorial on simpleExchange
    - Tutorial on devnet deployment

## Request: 
We intend to implement the best practices so the new developers can use them as a guide on how to properly build their components. By having this in mind, I would like to ask for your feedback on what are the current standards and what should be improved.
Also, there are some open questions that I would be very thankful if you could provide an answer.

## Open Questions:
- Contracts
    - Should we keep `swap and satisfy` methods of `@agoric/zoe/src/contractSupport`
    - Should we add any method to the `creatorFacet`
- UpgradableSimpleExchange
    - Should we use the `prepare` or the `zone` API
    - In what scenarios should we use the `prepareExo()` and the `prepareExoClass()`
- Devnet deployment
    - should we use `agoric run <proposal builder>` [#8087](https://github.com/Agoric/agoric-sdk/discussions/8087#discussioncomment-6534378)
    - version to use: agoric-sdk 0.21.2-u11.0
- Component page
    - should the focus be on the upgradable or non-upgradable version
- Tutorials
    - Video recording
- Best practices
    - Contract Code
    - In line comments
    - Tests
        - Structure
        - Tools
        - Coverage

**Comment by ivanlei** [comment]
Clsoing as this conversation has been redirected to the Agoric OpCo teams & discord.
